New monitor kiosk sessions using RUM guide (#18926)
DataDog/documentation: content/en/real_user_monitoring/guide/monitor-kiosk-sessions-using-rum.md

---
title: Monitor Kiosk Sessions Using RUM
kind: guide
description: Guide for monitoring kiosk sessions with RUM.
further_reading:
- link: '/real_user_monitoring/dashboards/'
  tag: 'Documentation'
  text: 'RUM Dashboards'

---

## Overview
Kiosk applications, such as fast food order machines and airline check-in terminals, often serve multiple users in quick succession. For that reason, triggering the end of a session based on a user action instead of waiting for an automatic session expiration (such as 15 minutes of inactivity or 4 hours total duration) is vital for collecting accurate session data and metrics for each user. With Datadog RUM SDKs, you can use this capability for an improved session tracking experience.

## Use `stopSession()` when users end their interaction

> Use the SDK `stopSession()` method to stop the session when the user finishes their interaction with the application, like when returning to the home screen or logging out, for example. A new session is created as soon as a user interacts with the application again or when a new View is started (mobile only).

Reviewer: ```suggestion
Use the SDK `stopSession()` method to stop the session when the user finishes their interaction with the application, like when returning to the home screen or logging out. A new session is created as soon as a user interacts with the application again or when a new View is started (mobile only).
```

> If a user was identified within the session, you might want to clear user information after calling `stopSession()` to start afresh. See documentation based on your application's framework: [Browser][1], [iOS][2], [Android][3], [Flutter][4], [React Native][5].

Reviewer: ```suggestion
If a user is identified within the session, you may want to clear user information after calling `stopSession()` to start afresh. See documentation based on your application's framework: [Browser][1], [iOS][2], [Android][3], [Flutter][4], [React Native][5].
```
